src/context/virtual-drive/shared/domain/DelayQueue.ts

```diff
@@ -20,26 +20,29 @@ export class DelayQueue {
     }
   }
 
-  private setTimeout() {
+  private setTimeout(delay = DelayQueue.DELAY) {
     this.clearTimeout();
     this.timeout = setTimeout(async () => {
       Logger.debug('Will try to run delay queue for: ', this.name);
       if (this.canLoop()) {
         Logger.debug('Running delay queue for: ', this.name);
-
         const reversedItems = Array.from(this.queue.entries()).reverse();
-
-        for (const [item] of reversedItems) {
-          await this.fn(item);
-          this.queue.delete(item);
+        await this.executeFn(reversedItems[0][0]);
+        if (!this.isEmpty) {
+          Logger.debug('Restarting delay queue is not empty');
+          this.setTimeout(500);
         }
-
         return;
       }
 
       Logger.debug(this.name, 'delay queue blocked');
       this.setTimeout();
-    }, DelayQueue.DELAY);
+    }, delay);
+  }
+
+  private async executeFn(item: string) {
+    this.queue.delete(item);
```

Comment:
Shouldn't be the other way around, first execute the function and delete it from the queue? I'm worried that if the deletion takes too long it can allow to delete files that does not need to be deleted